massa-execution-worker/src/execution.rs

```diff
@@ -1260,15 +1381,49 @@ impl ExecutionState {
             self.mip_store.clone(),
         );
 
+        // Deferred calls
+        let calls = execution_context.deferred_calls_advance_slot(
+            slot.clone(),
+            DEFERRED_CALL_MAX_FUTURE_SLOTS,
+            self.config.thread_count,
+        );
+
         // Get asynchronous messages to execute
         let messages = execution_context.take_async_batch(
-            self.config.max_async_gas,
+            self.config.max_async_gas.saturating_sub(calls.slot_gas),
             self.config.async_msg_cst_gas_cost,
         );
 
         // Apply the created execution context for slot execution
         *context_guard!(self) = execution_context;
 
+        for (id, call) in calls.slot_calls {
+            if call.cancelled {
+                // Skip cancelled calls
+                continue;
+            }
+            match self.execute_deferred_call(call) {
+                Ok(_exec) => {
+                    info!("executed deferred call: {:?}", id);
+                    cfg_if::cfg_if! {
+                        if #[cfg(feature = "execution-trace")] {
+                            // Safe to unwrap
+                            slot_trace.deferred_call_stacks.push(_exec.traces.unwrap().0);
+                        } else if #[cfg(feature = "execution-info")] {
+                            slot_trace.deferred_call_stacks.push(_exec.traces.clone().unwrap().0);
+                            exec_info.deferred_calls_messages.push(Ok(_exec));
+                        }
+                    }
+                }
+                Err(err) => {
+                    let msg = format!("failed executing deferred call: {}", err);
+                    #[cfg(feature = "execution-info")]
+                    exec_info.deferred_calls_messages.push(Err(msg.clone()));
+                    debug!(msg);
+                }
+            }
+        }
```

Comment:
are deferred calls deleted from db once executed ?

Reply:
also cancelled ones once scanned

Reply:
They are deleted in `advance_slot()`